Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/Views/PropertiesView.cs

```diff
@@ -1,30 +1,121 @@
         private NameAndParametersListView m_InteractionsListView;
         private NameAndParametersListView m_ProcessorsListView;
 
+        private Foldout interactionsFoldout => m_Root.Q<Foldout>("interactions-foldout");
+        private Foldout processorsFoldout => m_Root.Q<Foldout>("processors-foldout");
+
+        private TextElement addInteractionButton;
+        private TextElement addProcessorButton;
+
         public PropertiesView(VisualElement root, StateContainer stateContainer)
             : base(stateContainer)
         {
             m_Root = root;
 
             CreateSelector(
-                Selectors.GetSelectedBinding,
                 Selectors.GetSelectedAction,
+                Selectors.GetSelectedBinding,
                 state => state.selectionType,
-                (_, _, selectionType, _) => selectionType);
+                (inputAction, inputBinding, selectionType, s) => new ViewState()
+                {
+                    selectionType = selectionType,
+                    serializedInputAction = inputAction,
+                    inputBinding = inputBinding,
+                    relatedInputAction = Selectors.GetRelatedInputAction(s)
+                });
+
+            var interactionsToggle = interactionsFoldout.Q<Toggle>();
+            interactionsToggle.AddToClassList("properties-foldout-toggle");
+            if (addInteractionButton == null)
+            {
+                addInteractionButton = CreateAddButton(interactionsToggle, "add-new-interaction-button");
+                new ContextualMenuManipulator(_ => {}){target = addInteractionButton, activators = {new ManipulatorActivationFilter(){button = MouseButton.LeftMouse}}};
+            }
+            var processorToggle = processorsFoldout.Q<Toggle>();
+            processorToggle.AddToClassList("properties-foldout-toggle");
+            if (addProcessorButton == null)
+            {
+                addProcessorButton = CreateAddButton(processorToggle, "add-new-processor-button");
+                new ContextualMenuManipulator(_ => {}){target = addProcessorButton, activators = {new ManipulatorActivationFilter(){button = MouseButton.LeftMouse}}};
+            }
+        }
+
+        private TextElement CreateAddButton(Toggle toggle, string name)
+        {
+            var addProcessorButton = new TextElement();
+            addProcessorButton.text = "+";
+            addProcessorButton.name = name;
+            addProcessorButton.AddToClassList("add-interaction-processor-button");
+            toggle.Add(addProcessorButton);
+            return addProcessorButton;
         }
 
-        public override void RedrawUI(SelectionType selectionType)
+        private void CreateContextMenuProcessor(string expectedControlType, SerializedProperty serializedProperty)
+        {
+            var processors = InputProcessor.s_Processors;
+            Type expectedValueType = null;
+            if (!string.IsNullOrEmpty(expectedControlType))
```

Comment:
When you have a value being set above and then a conditional to update it I always prefer folding these into one to set the value only once.

var data = conditional ? truevalue : falseValue;

Reply:
So here

Type expectedValueType = string.IsNullOrEmpty(expectedControlType) ? null : EditorInputControlLayoutCache.GetValueType(expectedControlType);